x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/MachineLearning.java

```diff
@@ -1298,6 +1298,11 @@ public void cleanUpFeature(
         ActionListener<ResetFeatureStateResponse.ResetFeatureStateStatus> finalListener) {
         logger.info("Starting machine learning feature reset");
 
+        if (enabled == false) {
```

Comment:
@droberts195 Could you or someone else from the Machine Learning team let me know what you think of this? The problem I ran into was that when the ML plugin is disabled, `SetResetActionModeRequest` and other such actions aren't provided by the plugin, so the client fails to execute. It seems to me that if ML is disabled, we can just skip everything below, but I'd like to be sure about that.

Reply:
Yes, I think you are correct.  It's a requirement that if ML is being used then the plugin must be enabled on all master-eligible nodes.  If this rule has been obeyed then all the code below would be no-ops.  And if the rule hasn't been obeyed then the cluster is into undefined behaviour territory and other things will likely be going wrong anyway.

I agree it's still work calling the super class method because somebody might have used ML, then stopped using and deleted all the jobs and datafeeds before disabling the plugin, yet still have a few indices left over, so it's nice to clean those up.